### PR TITLE
Convert landing page to zero-JS Next.js application.

### PR DESCRIPTION
**Pull request by leerob**
This PR converts the custom React SSR landing page for bun.sh to a static, zero JavaScript Next.js application.

You can try it out here -> https://bun-landing-one.vercel.app/

### CODE REVIEW

File: packages/bun-landing/components/CodeBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import * as shiki from "shiki";
```
Comment (leerob) [inline packages/bun-landing/components/CodeBlock.tsx:1]: Refactored this out into a component.

File: packages/bun-landing/components/Layout.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                <li>
+                  <a className="NavText" href="https://github.com/oven-sh/bun">
+                    GitHub
+                  </a>
+                </li>
+              </ul>
+            </nav>
+          </header>
+        </div>
+        {children}
```
Comment (leerob) [inline packages/bun-landing/components/Layout.tsx:89]: Refactored out into a component to prepare for a future with more than one page.

File: packages/bun-landing/next.config.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+  reactStrictMode: true,
+  typescript: {
+    ignoreBuildErrors: true,
+  },
+  webpack: (config) => {
+    // support shiki top level await
+    config.experiments = { ...config.experiments, ...{ topLevelAwait: true }};
```
Comment (leerob) [inline packages/bun-landing/next.config.js:8]: Added to support top-level `await` for `shiki` – could refactor in the future.

File: packages/bun-landing/package.json [hunk 4]
```diff
@@ -1,9 +1,20 @@
...
-    "shiki": "^0.10.1"
+    "next": "12.2.3",
+    "react": "^18",
+    "react-dom": "^18",
+    "react-is": "^17.0.2",
+    "shiki": "0.10.1"
+  },
+  "devDependencies": {
+    "@types/react": "^18",
+    "@types/node": "latest",
```
Comment (leerob) [inline packages/bun-landing/package.json:16]: I needed this to get the build to pass – we might want to update the Next.js bun example to include instructions on running `bun run build` and make sure the TypeScript setup is correct. Can be a follow-up PR.

**Comment by leerob** [comment]
Note: We'll also want to update/remove `bun-landing.yml`

**Comment by Jarred-Sumner** [comment]
This is great. The only thing to change is make it so it uses bun to build the page instead of node.

For now, we can do that via a `bun dev` and then just saving the homepage with `curl` and inlining the assets into the HTML.

Bun will have a better solution to this later

I can go ahead and make that change

**Pull request merged**